pkg/api/pod/util.go

```diff
@@ -68,7 +92,7 @@ func VisitPodSecretNames(pod *api.Pod, visitor Visitor) bool {
 			return false
 		}
 	}
-	VisitContainers(&pod.Spec, func(c *api.Container) bool {
+	VisitContainers(&pod.Spec, DefaultContainers, func(c *api.Container, containerType ContainerType) bool {
```

Comment:
This is deciding for callers of VisitPodSecretNames that they should not visit containers disabled by feature gate.

A quick sweep of the uses of this method shows that containers that are feature-disabled but persisted should generally be included:
* kubelet secret manager: this should set up secret management for secret references from all persisted containers
* noderestriction admission: this should prevent referencing secrets in all containers
* serviceaccount admission: this should prevent mirror pods from referencing secrets in all containers
* node authorizer: this should honor persisted secret references

Reply:
Given their usage, we could change `VisitPodSecretNames` and `VisitConfigPodConfigmapNames` in this PR to visit all containers as a safer default and follow-up with a change to allow callers to signal whether they wish to obey the feature gate.

Reply:
yes, that sounds good.

Reply:
I think VisitConfigPodConfigmapNames is for VisitPodConfigmapNames which is covered by addressing comment below.